src/main/java/build/buf/protovalidate/CustomOverload.java

```diff
@@ -58,6 +57,11 @@ final class CustomOverload {
   private static final String OVERLOAD_IS_INF = "isInf";
   private static final String OVERLOAD_IS_HOST_AND_PORT = "isHostAndPort";
 
+  // See https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address
+  private static final Pattern EMAIL_REGEX =
+      Pattern.compile(
+          "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$");
```

Comment:
This expression is "JavaScript- and Perl-compatible". Is this expression supported by `java.util.regex.Pattern`?

Reply:
This looks like supported syntax to me - earlier versions of Java didn't have some features (like named capture groups) but later ones added that.

Reply:
Yes, I tested this on regex101. Plus it builds and passes conformance and unit tests.